src/checksums.ts

```diff
@@ -1,14 +1,40 @@
 import * as httpm from 'typed-rest-client/HttpClient'
 
+import fileWrapperChecksums from './wrapper-checksums.json'
+
 const httpc = new httpm.HttpClient(
   'gradle/wrapper-validation-action',
   undefined,
   {allowRetries: true, maxRetries: 3}
 )
 
+function getKnownValidChecksums(): Map<string, Set<string>> {
+  const versionsMap = new Map<string, Set<string>>()
+  for (const entry of fileWrapperChecksums) {
+    const checksum = entry.checksum
+
+    let versionNames = versionsMap.get(checksum)
+    if (versionNames === undefined) {
+      versionNames = new Set()
+      versionsMap.set(checksum, versionNames)
+    }
+
+    versionNames.add(entry.version)
+  }
+
+  return versionsMap
+}
+
+/**
+ * Known checksums from previously published Wrapper versions.
+ *
+ * Maps from the checksum to the names of the Gradle versions whose wrapper has this checksum.
+ */
+export const KNOWN_VALID_CHECKSUMS = getKnownValidChecksums()
```

Comment:
Can you add a unit test that verifies that this is never empty, and always contains some reasonable versions as a spot-check that this logic never breaks?